tests/test_sshfsmount.cpp

```diff
@@ -188,161 +239,152 @@ struct SshfsMount : public mp::test::SftpServerTest
 
     const std::unordered_map<std::string, std::string> default_cmds{
         {"snap run multipass-sshfs.env", "LD_LIBRARY_PATH=/foo/bar\nSNAP=/baz\n"},
-        {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 3.0.0"},
-        {"id -u", "1000"},
-        {"id -g", "1000"},
-        {"pwd", "/home/ubuntu"},
+        {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 3.0.0\n"},
+        {"id -u", "1000\n"},
+        {"id -g", "1000\n"},
+        {"pwd", "/home/ubuntu\n"},
         {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -o slave -o transform_symlinks -o allow_other :\"source\" "
          "\"target\"",
-         "don't care"}};
+         "don't care\n"}};
 };
-} // namespace
 
-TEST_F(SshfsMount, throws_when_sshfs_does_not_exist)
+// Mocks an incorrect return from a given command.
+struct SshfsMountFail : public SshfsMount, public testing::WithParamInterface<std::string>
 {
-    bool invoked{false};
-    auto request_exec = make_exec_that_fails_for({"sudo multipass-sshfs.env", "which sshfs"}, invoked);
-    REPLACE(ssh_channel_request_exec, request_exec);
+};
 
-    EXPECT_THROW(make_sshfsmount(), mp::SSHFSMissingError);
-    EXPECT_TRUE(invoked);
-}
+// Mocks correct execution of a given vector of commands/answers. The first string specifies the parameter with
+// which make_sshfsmount mock is called.
+struct SshfsMountExecute : public SshfsMount, public testing::WithParamInterface<std::pair<std::string, CommandVector>>
+{
+};
 
-TEST_F(SshfsMount, throws_when_unable_to_make_target_dir)
+// Mocks the server raising std::invalid_argument exception on the execution of given commands.
+struct SshfsMountInvalidArg : public SshfsMount, public testing::WithParamInterface<CommandVector>
 {
-    bool invoked{false};
-    auto request_exec = make_exec_that_fails_for({"mkdir"}, invoked);
-    REPLACE(ssh_channel_request_exec, request_exec);
+};
 
-    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
-    EXPECT_TRUE(invoked);
-}
+} // namespace
+
+//
+// Define some parameterized test fixtures.
+//
 
-TEST_F(SshfsMount, throws_when_unable_to_chown)
+TEST_P(SshfsMountFail, test_failed_invocation)
 {
-    bool invoked{false};
-    auto request_exec = make_exec_that_fails_for({"chown"}, invoked);
+    CommandVector commands = {
+        {"snap run multipass-sshfs.env", "LD_LIBRARY_PATH=/foo/bar\nSNAP=/baz\n"},
+        {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 3.0.0\n"},
+        {"pwd", "/home/ubuntu\n"},
+        {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+         "/home/ubuntu/\n"}};
```

Comment:
These should all go into `default_cmds` - we don't want `make_exec_that_executes_and_fails` to be checking the command ordering, do we?

I mean, _if_ we had a case where one command failing would just mean a different branch being taken, we'd need for each command in this vector to have a "fails" boolean member. But we don't have a case like that today.

Reply:
Right. In any case, if we really want to trace the execution of commands, we won't use `default_cmds`, but a `CommandVector` containing at least the ordering. Or, we can think on a better data structure to hold booleans or whatever for each command being executed. In any case, not using the default set of commands in `default_cmds`.